libs/top_level_metrics.py

```diff
@@ -46,12 +47,11 @@ def calculate_top_level_metrics_for_timeseries(timeseries: TimeseriesDataset, la
 
     top_level_metrics_data = {
         MetricsFields.CASE_DENSITY: case_density,
+        CommonFields.DATE: data[CommonFields.DATE],
         MetricsFields.TEST_POSITIVITY: test_positivity,
         CommonFields.FIPS: fips,
-        CommonFields.DATE: data[CommonFields.DATE],
     }
-
-    return pd.DataFrame(top_level_metrics_data).replace({np.nan: None}).set_index(CommonFields.DATE)
+    return pd.DataFrame(top_level_metrics_data, index=test_positivity.index).replace({np.nan: None})
```

Comment:
add comment about why np.nan is replaced with None. my hunch is that forces the column to be changed from float to object, which seems like it'd hurt performance.